### PR TITLE
BIP39 seed generator and switch android architecture to armeabi-v7a in debug

### PR DESCRIPTION
**Pull request by Rizvanov-Rinat**

### CODE REVIEW

File: electrum/mnemonic.py [hunk 1]
```diff
@@ -187,37 +187,36 @@ def mnemonic_decode(self, seed):
...
-            if is_old_seed(seed):
-                continue
-            # Make sure the mnemonic we generate is not also a valid bip39 seed
-            # by accident. Note that this test has not always been done historically,
-            # so it cannot be relied upon.
-            if bip39_is_checksum_valid(seed, wordlist=self.wordlist) == (True, True):
-                continue
-            if is_new_seed(seed, prefix):
-                break
+        seed = self.entropy_to_seed(self.random_bytes(num_bits//8), self.wordlist)
```
Comment (progbeat) [inline electrum/mnemonic.py]: Is `self.worldlist` BIP39 wordlist?

File: electrum/mnemonic.py [hunk 2]
```diff
@@ -187,37 +187,36 @@ def mnemonic_decode(self, seed):
...
-            i = entropy + nonce
-            seed = self.mnemonic_encode(i)
-            if i != self.mnemonic_decode(seed):
-                raise Exception('Cannot extract same entropy from mnemonic!')
-            if is_old_seed(seed):
-                continue
-            # Make sure the mnemonic we generate is not also a valid bip39 seed
-            # by accident. Note that this test has not always been done historically,
-            # so it cannot be relied upon.
-            if bip39_is_checksum_valid(seed, wordlist=self.wordlist) == (True, True):
```
Comment (progbeat) [inline electrum/mnemonic.py]: Check that `seed` is valid bip39 seed.

Comment (progbeat) [reply electrum/mnemonic.py]: Add asserts from seedgen to check that `entropy_to_seed()` is correct.

File: electrum/mnemonic.py (hunk 1 above)
Comment (Rizvanov-Rinat) [reply electrum/mnemonic.py]: Yes, it is.

File: electrum/mnemonic.py (hunk 2 above)
Comment (Rizvanov-Rinat) [reply electrum/mnemonic.py]: Added.

File: electrum/mnemonic.py [hunk 3]
```diff
@@ -187,38 +187,41 @@ def mnemonic_decode(self, seed):
             i = i*n + k
         return i
 
+    def random_bytes(self, num):
+        from time import time_ns
+        hasher = hashlib.sha3_512(os.getrandom(num))
+        hasher.update(time_ns().to_bytes(16, 'big'))
```
Comment (progbeat) [inline electrum/mnemonic.py:194]: Just in case also add:
```python
hasher.update(secrets.token_bytes(num))
```

File: electrum/mnemonic.py (hunk 2 above)
Comment (progbeat) [reply electrum/mnemonic.py:217]: ^^^
> Check that seed is valid bip39 seed.

File: electrum/mnemonic.py (hunk 3 above)
Comment (Rizvanov-Rinat) [reply electrum/mnemonic.py:194]: Added.

File: electrum/mnemonic.py [hunk 4]
```diff
@@ -187,38 +187,43 @@ def mnemonic_decode(self, seed):
...
+            words.append(wordlist[idx])
+        return ' '.join(words)
+
     def make_seed(self, *, seed_type=None, num_bits=None) -> str:
         from .keystore import bip39_is_checksum_valid
         if seed_type is None:
             seed_type = 'segwit'
         if num_bits is None:
-            num_bits = 132
+            num_bits = 256
```
Comment (progbeat) [inline electrum/mnemonic.py]:
```python
if seed_type == 'bip39':
    # your code
    return seed
# old code

File: electrum/mnemonic.py [hunk 5]
```diff
@@ -187,38 +187,43 @@ def mnemonic_decode(self, seed):
...
+    def random_bytes(self, num):
+        from time import time_ns
+        from secrets import token_bytes
+        hasher = hashlib.sha3_512(os.getrandom(num))
+        hasher.update(time_ns().to_bytes(16, 'big'))
+        hasher.update(token_bytes(num))
+        assert num <= hasher.digest_size
+        return hasher.digest()[:num]
+
+    def entropy_to_seed(self, entropy, wordlist):
```
Comment (progbeat) [inline electrum/mnemonic.py]: Use `self.wordlist`.
```suggestion
    def entropy_to_seed(self, entropy):
```

**Review by progbeat — APPROVED** [review]
LGTM

**Pull request merged**